File: Languages/Go/Day027.go
// Languages - Go - Day 27
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that counts how many vowels are in a given string.


package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
	"unicode"
)

// Function to count vowels in a given string using map for faster lookup
func countVowels(input string) int {
	vowels := map[rune]bool{'a': true, 'e': true, 'i': true, 'o': true, 'u': true}
	count := 0

	for _, char := range input {
		if vowels[unicode.ToLower(char)] {
			count++
		}
	}
	return count
}

func main() {
	reader := bufio.NewReader(os.Stdin)

	for {
		fmt.Println("Enter a string to count the number of vowels (or type 'exit' to quit):")
		input, _ := reader.ReadString('\n')
		input = strings.TrimSpace(input)

		if strings.ToLower(input) == "exit" {
			fmt.Println("Exiting program. Goodbye!")
			break
		}

		// Input validation
		if input == "" {
			fmt.Println("Invalid input. Please enter a non-empty string.")
			continue
		}

		vowelCount := countVowels(input)
		fmt.Printf("The number of vowels in the given string is: %d\n", vowelCount)
	}
}

File: Languages/C#/Day027.cs
// Languages - C# - Day 27
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that counts how many vowels are in a given string


using System;
using System.Collections.Generic;

class Program
{
    // Function to count vowels in a given string
    static int CountVowels(string input)
    {
        HashSet<char> vowels = new HashSet<char> { 'a', 'e', 'i', 'o', 'u' };
        int count = 0;

        foreach (char ch in input.ToLower())
        {
            if (vowels.Contains(ch))
            {
                count++;
            }
        }
        return count;
    }

    static void Main()
    {
        while (true)
        {
            Console.WriteLine("Enter a string to count the number of vowels (or type 'exit' to quit):");
            string input = Console.ReadLine();

            if (string.IsNullOrEmpty(input))
            {
                Console.WriteLine("Invalid input. Please enter a non-empty string.");
                continue;
            }

            if (input.ToLower() == "exit")
            {
                Console.WriteLine("Exiting program. Goodbye!");
                break;
            }

            int vowelCount = CountVowels(input);
            Console.WriteLine($"The number of vowels in the given string is: {vowelCount}");
        }
    }
}
